cpp/src/arrow/compute/exec/filter_node.cc

```diff
@@ -99,11 +105,35 @@ class FilterNode : public ExecNode {
   void InputReceived(ExecNode* input, ExecBatch batch) override {
     DCHECK_EQ(input, inputs_[0]);
 
-    auto maybe_filtered = DoFilter(std::move(batch));
-    if (ErrorIfNotOk(maybe_filtered.status())) return;
+    ARROW_LOG(DEBUG) << "FilterNode: >> input";
+
+    auto executor = plan()->exec_context()->executor();
+    if (executor) {
+      auto maybe_future = executor->Submit([this, batch] {
```

Comment:
Whatever we submit to the executor must be more than just a lambda if it is to encode information that the scheduler needs to use. A cost function and context about what query this belongs to and other metadata might want to be part of what the task includes.

Reply:
I am not saying that we need to include those things now but it would be good to see if there is a way to send this information with the task

Reply:
done